service/frontend/workflowHandler.go

```diff
@@ -2086,10 +2086,16 @@ func (wh *WorkflowHandler) ResetWorkflowExecution(ctx context.Context, request *
 	if request == nil {
 		return nil, errRequestNotSet
 	}
-
 	if request.GetNamespace() == "" {
 		return nil, errNamespaceNotSet
 	}
+	if request.GetRequestId() == "" {
```

Comment:
How about if request ID not set, service should generate a random uuid?

Reply:
RequestID exists to dedup in case of client side retry. If we generate uuid on server, then it lost its meaning.

Reply:
no, if request ID is provided, then use the request ID.
if request ID not provided, the we rand a UUID as request ID.

Reply:
That could be an option, but it might result in unexpected result and cause surprise to user. 
Also, our StartWorkflowExecution has the same RequestId and it does the same empty check and return the same error. Better keep consistent behavior.

Reply:
to me, request ID is similar to page size, this kind of variables / inputs are optional.
let's check with other folks.

Reply:
this is kind of thing @mfateev  will have an opinion on

Reply:
Generate a UUID in case of empty would be a desired behavior on SDK side before retry logic. SDK-go does that. cc @Spikhalskiy

Reply:
@yiminc I agree. But user has been using a raw gRPC in the original user request. So, this whole situation is about server API only.

Reply:
I'm with @wxing1292  here. I think that server should generate a randomId or not dedup an empty requestId. 
That would be an expected behavior from my point of view.
A lot of idempotent apis make idempotency key optional and they just don't enforce idempotency if it's not provided.

Reply:
@Spikhalskiy you proposed 3 options in #2140, 1) make RequestId mandatory; 2) gen UUID on empty RequestId; 3) error out on empty RequestId. 1) and 3) are the same thing and is what this PR implemented. You and @wxing1292 prefer 2). 
Treat idempotency key as required or not depends on use case regarding duplication is allowed or not. In case of StartWorkflow and ResetWorkflow, gen a UUID on empty RequestId won't work because we only allow one running workflow for any given WorkflowID. Assuming client side does not set RequestId and the call timed out, so client do retry. If we generate UUID on empty RequestId, all retry will fail because we cannot start duplicate workflow. On the other hand, for use case where duplicate is allowed like SignalWorkflowExecution then server could optionally handle dedup only if RequestId is present. In fact, this is the behavior of SignalWorkflow. If you don't provide RequestId, server just don't do dedup. 
So I think it is better to enforce RequestId here for ResetWorkflow and keep consistent behavior with StartWorkflow.

Reply:
^ then request ID has different behavior: signal vs start & reset
our APIs should have similar behavior.